### PR TITLE
`Exam mode`: Fix setting an individual student exam working time

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context
Bug noticed by @FlorianGlombik in https://github.com/ls1intum/Artemis/pull/4535#pullrequestreview-838805920.

### Description
The current form saves the working time in minutes, but it is actually in seconds. Therefore, the displayed duration was wrong.

### Steps for Testing
Prerequisites:
- 1 Exam

1. Go to a student exam.
2. Change the working time. It should be displayed correctly after the update as hours and minutes.
3. Optional: Also check the network traffic when refreshing the page for the JSON with the student exam (API-endpoint `/api/courses/$courseId/exams/$examId/student-exams/$studentExamId`). The `workingTime` field there contains the total number of seconds of working time. This should match the values displayed in the UI.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage
unchanged

### Screenshots
Screenshot 2022-01-03 at 21-02-22 Exam Management

### CODE REVIEW

**Review by FlorianGlombik — CHANGES_REQUESTED** [review]
In the student exams review the `Working time` should be displayed in a different way.
It is still shown in the `minutes` and `seconds` format.

image

**Comment by b-fein** [comment]
> In the student exams review the `Working time` should be displayed in a different way in the `Student exams` view. It is still shown in the `minutes` and `seconds` format.

@FlorianGlombik Thanks for the review.
All the other dates displayed are done so by using the duration pipe. Therefore, the seconds are shown as well. I now updated the form when setting the working time to enable setting the seconds as well to stay consistent.
Before #4234 the duration could be set in minutes and seconds. This ability has now been added back in with the additional bonus of conveniently setting (the probably more frequently needed) hours.

**Comment by krusche** [comment]
Can we add a second option to edit the working time? In our university, students typically get an extension in % of the normal working time, e.g. 50%. It would be easier for the instructor to enter this value. The correct working time could then be calculated automatically based on the percentage value. 

We should offer both ways next to each other. Changing one value will automatically update the other value. In the database, we finally store the time and not the percentages, so no change of the database would be necessary.

This percentage value could then also be displayed on other pages, e.g. in the student exams page after the working time, e.g. when the normal working time is 30min.

- 45 min (+ 50%)
- 30 min
- 48 min (+ 60%)
- 15 min (- 50%)

(the last one is only a theoretical example, which we should support, but which will probably never occur in reality)

Could you add this in this PR? Or should we add it in another one?

**Comment by b-fein** [comment]
> Can we add a second option to edit the working time? In our university, students typically get an extension in % of the normal working time, e.g. 50%. It would be easier for the instructor to enter this value. The correct working time could then be calculated automatically based on the percentage value.
> 
> We should offer both ways next to each other. Changing one value will automatically update the other value. In the database, we finally store the time and not the percentages, so no change of the database would be necessary.
> 
> This percentage value could then also be displayed on other pages, e.g. in the student exams page after the working time, e.g. when the normal working time is 30min.
> 
>     * 45 min (+ 50%)
> 
>     * 30 min
> 
>     * 48 min (+ 60%)
> 
>     * 15 min (- 50%)
> 
> 
> (the last one is only a theoretical example, which we should support, but which will probably never occur in reality)
> 
> Could you add this in this PR? Or should we add it in another one?

I’d like this PR to only contain the bugfix for the broken input and then add the new feature separately, as at the moment the functionality to set individual working times for student exams is broken completely. That way the bugfix can be merged as soon as possible.

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html [hunk 1]
```diff
@@ -37,8 +37,9 @@ <h5><span jhiTranslate="artemisApp.studentExamDetail.student">Student</span></h5
         <h5><span jhiTranslate="artemisApp.studentExams.workingTime" [ngbTooltip]="getWorkingTimeToolTip()">Working time</span></h5>
         <form class="d-flex" [formGroup]="workingTimeForm" (ngSubmit)="saveWorkingTime()">
             <div class="me-2" [ngbTooltip]="getWorkingTimeToolTip()">
-                <input formControlName="minutes" type="number" style="width: 60px; height: 35px" class="text-center" min="0" step="1" required /> h
-                <input formControlName="seconds" type="number" style="width: 60px; height: 35px" class="text-center" min="0" max="59" step="1" required /> min
+                <input formControlName="hours" type="number" style="width: 60px; height: 35px" class="text-center" min="0" step="1" required /> h
+                <input formControlName="minutes" type="number" style="width: 60px; height: 35px" class="text-center" min="0" max="59" step="1" required /> min
+                <input formControlName="seconds" type="number" style="width: 60px; height: 35px" class="text-center" min="0" max="59" step="1" required /> s
```
Comment (FlorianGlombik) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html:42]: It might make sense to introduce a `css` class for the style? 
e.g. 
```
.durationInput {
    width: 60px;
    height: 35px;
}
```

```suggestion
                <input formControlName="hours" type="number" class="text-center durationInput " min="0" step="1" required /> h
                <input formControlName="minutes" type="number" class="text-center durationInput " min="0" max="59" step="1" required /> min
                <input formControlName="seconds" type="number" class="text-center durationInput " min="0" max="59" step="1" required /> s
```

I think that it would be better to use bootstrap input fields instead of styling raw input fields, but from a short look we are using raw input fields within the application so far - so if we would want to change this it would be another PR.

File: src/main/webapp/app/shared/pipes/artemis-duration-from-seconds.pipe.ts [hunk 2]
```diff
@@ -1,5 +1,12 @@
 import { Pipe, PipeTransform } from '@angular/core';
 
+interface Duration {
```
Comment (FlorianGlombik) [inline src/main/webapp/app/shared/pipes/artemis-duration-from-seconds.pipe.ts:3]: Isn't this rather a `type` than an interface?

**Review by FlorianGlombik — APPROVED** [review]
Tested it on ts3, I was able to set the duration for a student exam and it was displayed properly. 

In my test exam there was one programming exercise, the repositories were locked properly, the left time was displayed properly (tested with 3 different times, all longer than the actual time).

I did not find issues within the code, I only left a few small comments.

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html (hunk 1 above)
Comment (b-fein) [reply src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.html:42]: I think I’m going to change this form completely anyway in the implementation for #4581 to be able to make the value display easily dynamic when changing either the percent or the duration values. So I’m not going to change it here, but I’ll keep it in mind for the next PR.

At the moment I think that something like in the [Bootstrap documenation for input groups](https://getbootstrap.com/docs/5.1/forms/input-group/#basic-example) for the ‘Username @ Server’ would look nice to group the Xh Ymin Zsec inputs.

File: src/main/webapp/app/shared/pipes/artemis-duration-from-seconds.pipe.ts (hunk 2 above)
Comment (b-fein) [reply src/main/webapp/app/shared/pipes/artemis-duration-from-seconds.pipe.ts]: Yes you’re right. It groups values, not methods. Changed it in 1c38c11ec1d9e605f011a08438a9dff0a51cfed3.

**Review by FlorianGlombik — APPROVED** [review]
Re-approve as only the `interface` was changed to `type` in https://github.com/ls1intum/Artemis/pull/4548/commits/1c38c11ec1d9e605f011a08438a9dff0a51cfed3 since the last review
